### PR TITLE
perf: Where clause subquery optimization

### PR DESCRIPTION
**Pull request by timgl**
## Problem

#25718 didn't speed up ActorsQuery searching yet because whereclauseextractor didn't optimize if the query had a subquery

## Changes

Add subquery support to whereclauseextractor

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## How did you test this code?

This reverts commit 6b91cb8.

### CODE REVIEW

File: posthog/hogql/database/schema/util/where_clause_extractor.py [hunk 1]
```diff
@@ -120,20 +120,25 @@ def visit_compare_operation(self, node: ast.CompareOperation) -> ast.Expr:
...
             left = self.visit(node.left)
-            right = self.visit(node.right)
 
-            # Too complicated
             if isinstance(node.right, ast.SelectQuery):
-                return ast.Constant(value=True)
+                right = clone_expr(
+                    node.right, clear_types=False, clear_locations=False, inline_subquery_field_names=True
+                )
+                # right = self.visit(node.right)
```
Comment (mariusandra) [inline posthog/hogql/database/schema/util/where_clause_extractor.py:128]: worth removing commented out code

File: posthog/hogql/database/schema/util/where_clause_extractor.py [hunk 2]
```diff
@@ -120,20 +120,25 @@ def visit_compare_operation(self, node: ast.CompareOperation) -> ast.Expr:
...
 
         return ast.Constant(value=True)
 
-    def visit_select_query(self, node: ast.SelectQuery) -> ast.Expr:
-        # going too deep, bail
-        return ast.Constant(value=True)
+    # def visit_select_query(self, node: ast.SelectQuery) -> ast.Expr:
+    #     import ipdb;ipdb.set_trace()
+    #     # going too deep, bail
+    #     return ast.Constant(value=True)
```
Comment (mariusandra) [inline posthog/hogql/database/schema/util/where_clause_extractor.py:141]: same here

**Review by mariusandra — APPROVED** [review]
Small nits, otherwise LGTM!

**Comment by timgl** [comment]
superseded by 25804

**Pull request closed without merging**